models/usage.go

```diff
@@ -0,0 +1,244 @@
+}
+
+// GetUploadPricePerGB is used to get the upload price per gb for a user
+// allows us to specify whether the payment
+func (bm *UsageManager) GetUploadPricePerGB(username string) (float64, error) {
+	b, err := bm.FindByUserName(username)
+	if err != nil {
+		return 0, err
+	}
+	return b.Tier.PricePerGB(), nil
+}
+
+// CanPublishIPNS is used to check if a user can publish IPNS records
+func (bm *UsageManager) CanPublishIPNS(username string) (bool, error) {
+	b, err := bm.FindByUserName(username)
+	if err != nil {
+		return false, err
+	}
+	// if they are free, make sure they aren't at the limit
+	if b.Tier == Free && b.IPNSRecordsPublished > 5 {
+		return false, nil
+	}
+	return true, nil
+}
+
+// CanPublishPubSub is used to check if a user can publish pubsub messages
+func (bm *UsageManager) CanPublishPubSub(username string) (bool, error) {
+	b, err := bm.FindByUserName(username)
+	if err != nil {
+		return false, err
+	}
+	if b.Tier == Free && b.IPNSRecordsPublished > 10000 {
+		return false, nil
+	}
+	return true, nil
+}
+
+// UpdateDataUsage is used to update the users' data usage amount
+// If the account is non free, and the upload pushes their total monthly usage
+// above the tier limit, they will be upgraded to the next tier to receive the discounted price
+// the discounted price will apply on subsequent uploads.
+// If the 1TB maximum monthly limit is hit, then we throw an error
+func (bm *UsageManager) UpdateDataUsage(username string, uploadSize float64) error {
+	b, err := bm.FindByUserName(username)
+	if err != nil {
+		return err
+	}
+	// update size
+	b.CurrentDataUsedGB = b.CurrentDataUsedGB + uploadSize
+	if b.Tier == Free {
+		// if they are free, they will need to upgrade their plan
+		if b.CurrentDataUsedGB >= datasize.GB.GBytes()*3 {
+			return errors.New("upload limit will be reached, please upload smaller content or upgrade your plan")
+		}
+	} else if b.Tier == Light {
+		// if they are light plan, and this upload takes them over 100GB
+		// upad their tier to plus to allow for free uploads
+		if b.CurrentDataUsedGB >= datasize.GB.GBytes()*100 {
+			// update tear
+			b.Tier = Plus
+		}
+	}
+	// check for the max upload limit of 1TB
+	if b.CurrentDataUsedGB >= datasize.TB.GBytes() {
+		return errors.New("max upload limit of 1TB reached, contact support")
+	}
+	// save updated columns and return
+	return bm.DB.Model(b).UpdateColumns(map[string]interface{}{
+		"tier":                 b.Tier,
+		"current_data_used_gb": b.CurrentDataUsedGB,
+	}).Error
+}
+
+// HasStartedPrivateNetworkTrial is used to check if the user has started their private network trial
+func (bm *UsageManager) HasStartedPrivateNetworkTrial(username string) (bool, error) {
+	b, err := bm.FindByUserName(username)
+	if err != nil {
+		return false, err
+	}
+	return b.PrivateNetworkTrialUsed, nil
+}
+
+// UpdateTier is used to update the Usage tier associated with an account
+func (bm *UsageManager) UpdateTier(username, tier string) error {
```

Comment:
```suggestion
 func (bm *UsageManager) UpdateTier(username string, tier DataUsageTier) error {
```

Reply:
Good catch, I also updated the switch/case to use the typed string